### PR TITLE
Miscellaneous improvements and bug fixes

### PR DESCRIPTION
**Pull request by Naveen3Singh**
This PR addresses issues https://github.com/SimpleMobileTools/Simple-Music-Player/issues/470, https://github.com/SimpleMobileTools/Simple-Music-Player/issues/391, https://github.com/SimpleMobileTools/Simple-Music-Player/issues/376, and some notification, widget related improvements and bug fixes. 

### Major changes:
 - Remove the fake notification as it led to seemingly unpredictable side effects.
 - Update foreground logic and move notification creation code to a separate helper class
 - Save and restore the current queue item and its position, previously everything was reset and service was stopped.
 - Update widgets properly on track state change, track info wasn't being updated.
 - Add a close (❌) button in the player notification on all android versions.
 - Save the state, pause the player and hide the notification when the sleep timer ends, no need to kill the service.
 - Move the media session callback object to a separate file, it'll grow in size later.

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/musicplayer/helpers/NotificationHelper.kt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                )
+                notificationChannel.enableLights(false)
+                notificationChannel.enableVibration(false)
+                notificationChannel.setShowBadge(false)
+
+                notificationManager.createNotificationChannel(notificationChannel)
+            }
+        }
+
+        fun from(
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/musicplayer/helpers/NotificationHelper.kt:148]: this doesnt seem to be clear enough, try using more descriptive function names

**Comment by tibbi** [comment]
seems to work nice, just found 1 issue with bluetooth initializing. If I have the app shut down and get into the car and it connects to my mobile, it automatically triggers KeyEvent.KEYCODE_MEDIA_PLAY, which is ok. The audio starts playing, but the current tracks name, progress etc arent shown neither in the car, nor in the mobile notification. The notification shows up with no info about the current track and the controls work, but even if I pause/play or do next song, the played files data doesnt show up anywhere. Can you somehow reproduce it?

**Comment by Naveen3Singh** [comment]
I'm working on it...
I don't immediately have access to a car but I can reproduce a similar issue on a Samsung phone running android 12, the title and artist data don't show up but it does right after I pause or hit next.

> even if I pause/play or do next song, the played files data doesnt show up anywhere

that's really strange, even if the metadata update was missed once, it should get updated at the next call. The "empty" notification is still there on the phone when this happens?

**Comment by Naveen3Singh** [comment]
> If I have the app shut down and get into the car and it connects to my mobile

does everything show up correctly when you have the app in the foreground?

**Comment by tibbi** [comment]
seems to work really well now, will see if other people report some issues :) So is this ready to be merged?

**Comment by tibbi** [comment]
I see that you havent really changed actual code since my last review, just some readability stuff. Guess those issues depend on some circumstances, we will see how will it behave in real world. I would merge it if ready.

**Comment by Naveen3Singh** [comment]
yep, it can be merged.

I have been using the app for playing music and audiobooks since I created the PR and found no issues on android 13. I encountered that empty metadata issue only one time while the app was in the background but I couldn't reproduce it again. I guess we'll address those issues another time in another PR :)

**Comment by Naveen3Singh** [comment]
I think #391 and #376 can be closed now.

**Pull request merged**